[DOCUMENTATION] Support use of IRSA for repository-s3 plugin credentials (#630)
opensearch-project/documentation-website: _opensearch/snapshot-restore.md

   s3.client.default.proxy.port: 8080 # port for your proxy server
   s3.client.default.read_timeout: 50s # the S3 connection timeout
   s3.client.default.use_throttle_retries: true # whether the client should wait a progressively longer amount of time (exponential backoff) between each successive retry
   s3.client.default.region: us-east-2 # AWS region to use
   ```

> 1. (Optional) Alternatively to AWS access and secret keys, you could configure plugin to use AWS IAM roles for service accounts:

Reviewer: I would switch to:
```suggestion
1. (Optional) If you don't want to use AWS access and secret keys, you could configure S3 plugin to use AWS IAM roles for service accounts:
```

> 1. (Optional) If you don't want to use AWS access and secret keys, you could configure S3 plugin to use AWS IAM roles for service accounts:

Reviewer 2: Insert "the" following "configure". If this is its first appearance, IAM should be defined as "AWS Identity and Access Management (IAM)". For subsequent appearances, use only "IAM" (do not use "AWS IAM").

Author: :+1:

   ```bash
   sudo ./bin/opensearch-keystore add s3.client.default.role_arn
   sudo ./bin/opensearch-keystore add s3.client.default.role_session_name
   ```

>    Optionally, if AWS access and secret keys are not configured, add to `opensearch.yml` setting (please make sure the file is accessible by `repository-s3` plugin):

Reviewer: We can remove "Optionally" since we already indicated the step itself is optional. Maybe switch to:

```suggestion
Alternatively, you can add your S3 token to your `opensearch.yml` settings. Make sure your token file is accessible by `repository-s3` plugin:
```

Author: This one seems to change the meaning of the sentence: essentially, either AWS access and secret keys or token file is needed, but not both.

Reviewer: Understood. Here's an alternative:

```suggestion
 If you don't want to configure AWS access and secret keys, modify the following `opensearch.yml` setting. Make sure the file is accessible by the `repository-s3` plugin:
```

   ```yml
   s3.client.default.identity_token_file: /usr/share/opensearch/plugins/repository-s3/token
   ```

>    AWS IAM roles require at least one of the above settings. Other settings will be taken from environment variables (if available): `AWS_ROLE_ARN`, `AWS_WEB_IDENTITY_TOKEN_FILE` and `AWS_ROLE_SESSION_NAME`.

Reviewer 2: Change "AWS IAM" to "IAM". Insert a comma after "'AWS_WEB_IDENTITY_TOKEN_FILE'".

>    For authenticating with AWS IAM roles for service accounts, at least one from the settings listed above has to be provided. Other settings will be taken from environment variables (if available): `AWS_ROLE_ARN`, `AWS_WEB_IDENTITY_TOKEN_FILE` and `AWS_ROLE_SESSION_NAME`.

Reviewer: ```suggestion
 AWS IAM roles require at least one of the above settings. Other settings will be taken from environment variables (if available): `AWS_ROLE_ARN`, `AWS_WEB_IDENTITY_TOKEN_FILE` and `AWS_ROLE_SESSION_NAME`.
```

Author: :+1: thanks!
